src/main/java/io/strimzi/kafka/bridge/http/converter/HttpJsonMessageConverter.java

```diff
@@ -30,6 +30,16 @@
 
 public class HttpJsonMessageConverter implements MessageConverter<String, byte[], Buffer, Buffer> {
 
+    private Integer partition;
+
+    public HttpJsonMessageConverter(int partition) {
+        this.partition = partition;
+    }
+
+    public HttpJsonMessageConverter() {
+        this.partition = null;
+    }
+
```

Comment:
Instead of "hiding" the partition in the ctor I would create and overload of the `toKafkaRecord` method (of course in the `MessageConverter` interface as well, updating the related AMQP converter too). This overload could get the `partition` (alongside `kafkaTopic` and `message`). I would use `Integer` so that one overload can call the other with `null` meaning that the partition is in the JSON `message`.